packages/parser/tests/unit/schema/kinesis.test.ts

```diff
@@ -110,4 +116,10 @@ describe('Kinesis ', () => {
       '49640912821178817833517986466168945147170627572855734274000000'
     );
   });
+
+  it('should catch any unknown fields in the example event', () => {
```

Comment:
I see where you're coming from with the addition of `.strict()` to the schemas, but I'm not sure it's the right choice in this situation.

The main concern I have is that we're no longer testing the model we're shipping to customers but instead a more strict version of it. Some customers might end up using `.strict()` on their own, which is actually a good idea, but I think it'd be better to test the schema as-is.

Ultimately, while the `.strict()` method does the job, we're really just observing a side effect rather than checking that the parsed output is what it is.

From a DX perspective, for us maintainers, using `.strict()` gives us an error like this:

<img width="624" alt="image" src="https://github.com/user-attachments/assets/3386a497-96bc-4e4e-b85b-c6abebfe6262">

Whereas checking the object would give us an actual diff of the object, clearly pointing at the missing field:

<img width="768" alt="Screenshot 2024-10-28 at 11 18 46" src="https://github.com/user-attachments/assets/e9b130b1-fe02-44ef-9546-d0975ba49d40">

While I recognise that the output is primarily a style choice, and both are serviceable, I think ultimately testing the output on the same schema we ship is more accurate.

For example, in this case you could modify the first test and check the entire payload instead of only one key, as well as changing the matcher from `toEqual()` to `toStrictEqual()`.

```ts
it('should parse kinesis event', () => {
  // Prepare
  const kinesisStreamEvent = TestEvents.kinesisStreamEvent;
  const expectedPayload = structuredClone(
    TestEvents.kinesisStreamEvent
  ) as KinesisDataStreamEvent;
  expectedPayload.Records[0].kinesis.data = 'Hello, this is a test.';
  expectedPayload.Records[1].kinesis.data = 'This is only a test.';

  // Act
  const parsed = KinesisDataStreamSchema.parse(kinesisStreamEvent);

  // Assess
  expect(parsed).toStrictEqual(expectedPayload);
});
```

This is also how I realized the key was missing while refactoring the tests in #3248.

Reply:
That's good, it means the other schemas were correct!